Sources/StreamChat/Workers/ChannelListUpdater.swift

```diff
@@ -135,6 +146,43 @@ class ChannelListUpdater: Worker {
             completion?($0.error)
         }
     }
+
+    /// Links a channel to the given query.
+    func link(channel: ChatChannel, with query: ChannelListQuery, completion: ((Error?) -> Void)? = nil) {
+        database.write { session in
+            guard let queryDTO = session.channelListQuery(filterHash: query.filter.filterHash) else {
+                log.debug("Channel list query has not yet created \(query)")
+                return
+            }
+
+            guard let channelDTO = session.channel(cid: channel.cid) else {
+                log.debug("Channel \(channel.cid) cannot be found in database.")
+                return
+            }
+
+            queryDTO.channels.insert(channelDTO)
+        } completion: { error in
+            completion?(error)
+        }
+    }
+
+    /// Unlinks a channel to the given query.
+    func unlink(channel: ChatChannel, with query: ChannelListQuery, completion: ((Error?) -> Void)? = nil) {
```

Comment:
We could probably have a private method with the shared logic of this 2 methods

Reply:
The duplication is minimal, but yes I think we can probably reuse most of the logic into a private function, will do it then 👍

Reply:
Done